### PR TITLE
[feat] add microbatch forwarding

### PR DESCRIPTION
**Pull request by YeAnbang**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [ ] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs
- [x] I have installed pre-commit: `pip install pre-commit && pre-commit install`

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [x] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: applications/ColossalChat/coati/distributed/launch.py [hunk 1]
```diff
@@ -101,7 +102,11 @@ def launch_distributed(
             plugin_config=plugin_config,
             microbatch_size=train_microbatch_size,
             generate_config=generate_config_consumer,
-            training_config={"filter_range": [0.05, 9.0], "lr": 1e-6},
+            training_config={
+                "filter_range": [0.05, 9.0],
```
Comment (TongLi3701) [inline applications/ColossalChat/coati/distributed/launch.py:106]: Why do we have 9 here ? I assume it should be between 0 - 1?

File: applications/ColossalChat/rl_example.py [hunk 2]
```diff
@@ -49,6 +49,12 @@
         )
     elif args.backend == "vllm":
         inference_model_config.update(dict(gpu_memory_utilization=0.7, enforce_eager=True, enable_chunked_prefill=True))
+        train_model_config.update(
+            dict(
+                use_flash_attention_2=True,
+                use_cache=False,
+            )
+        )
```
Comment (TongLi3701) [inline applications/ColossalChat/rl_example.py:57]: train config should be outside the backend part.

**Review by TongLi3701 — CHANGES_REQUESTED** [review]
Thanks Anbang, I left some comments. Please have a look.

File: applications/ColossalChat/coati/distributed/grpo_consumer.py [hunk 3]
```diff
@@ -131,40 +132,16 @@ def step(self, step_idx: int, **kwargs) -> Optional[float]:
         num_action = action_mask.shape[1]
         old_action_log_probs = data["action_log_probs"]
         response_length = torch.sum(action_mask, dim=1).to(torch.float32)
+        forward_batch_size = self.training_config.get("forward_micro_batch_size", data["input_ids"].size(0))
```
[the remark this answers is not in the document]
Comment (YeAnbang) [reply applications/ColossalChat/coati/distributed/grpo_consumer.py:135]: Not necessary, forward_batch_size can be set to any value, even greater than train bs, as long as you have enough memory

[the remark this answers is not in the document]
Comment (YeAnbang) [reply applications/ColossalChat/coati/distributed/grpo_consumer.py:135]: For other hyperparameters, we can add an assertion checking, but that requires knowing the distributed training strategy (tp, sp, pp).

File: applications/ColossalChat/coati/distributed/launch.py (hunk 1 above)
Comment (YeAnbang) [reply applications/ColossalChat/coati/distributed/launch.py:106]: The filtering is based on raw reward 0-10 in our case and should be given by the user

**Review by TongLi3701 — APPROVED** [review]
Thanks Anbang.
